src/sha2.c

```diff
@@ -921,8 +921,8 @@ void SHA512_Last(SHA512_CTX* context) {
 		*context->buffer = 0x80;
 	}
 	/* Store the length of input data (in bits): */
-	*(sha2_word64*)&context->buffer[SHA512_SHORT_BLOCK_LENGTH] = context->bitcount[1];
-	*(sha2_word64*)&context->buffer[SHA512_SHORT_BLOCK_LENGTH+8] = context->bitcount[0];
+	memcpy(&context->buffer[SHA512_SHORT_BLOCK_LENGTH], &context->bitcount[1], sizeof(*context->bitcount));
```

Comment:
Using `sizeof(*context->bitcount)` here is ok, although I wonder if it wouldn't be clearer if we used `sizeof(context->bitcount[1])` instead. Same for the line below (with index 0, of course). Otherwise, it all LGTM.

Reply:
Yeah, I had the same thought and I couldn't decide. I decided that "morally" it was cleaner to use what others had used, even if I disliked the style.

But you saying it tips me over the edge. Let me do that now.

Reply:
Done - thanks!